padl/transforms.py

```diff
@@ -1760,520 +1844,554 @@ def _add_name_to_splits(self, final_splits):
+                    self.connect(out_node,
+                                 self.output_node)
+                continue
+            node = Node(transform)
+            self.connect(self.input_node,
+                         node,
+                         output_slice=self.input_node.pd_output_slice,
+                         input_slice=node.pd_input_slice)
+            self.connect(node,
+                         self.output_node,
+                         output_slice=node.pd_output_slice,
+                         input_slice=self.output_node.pd_input_slice)
 
-        if Transform.pd_mode is not None:
-            return tuple(out)
-        return self._pd_output_formatter(out)
 
-    def _pd_longrepr(self, formatting=True, marker=None) -> str:
-        if not formatting:
-            make_green_ = lambda x: x
-            make_bold_ = lambda x: x
-        else:
-            make_green_ = make_green
-            make_bold_ = make_bold
-
-        def pipes(n):
-            return "│" * n
-
-        def spaces(n):
-            return " " * n
-
-        def horizontal(n):
-            return "─" * n
-        len_ = len(self.transforms)
-        out = ''
-        for i, t in enumerate(self.transforms):
-            out += (
-                make_green_(pipes(len_ - i - 1) + '└' + horizontal(i + 1) + '▶ ') +
-                make_bold_(f'{i}: ') + t._pd_shortrepr()
-            )
-            out += marker[1] + '\n' if marker and marker[0] == i else '\n'
-            if i < len(self.transforms) - 1:
-                out += f'{make_green_(pipes(len_ - i - 1) + spaces(i + 2) + self.display_op)}  \n'
+class Parallel(Pipeline):
+    op = '/'
+    _name = 'parallel'
 
-        return out
+    def __init__(self, transforms: Iterable[Transform], call_info: inspector.CallInfo = None,
+                 pd_name: Optional[str] = None, pd_group: bool = False):
+        super().__init__(transforms, call_info=call_info, pd_name=pd_name, pd_group=pd_group)
+
+        for idx, transform in enumerate(self.transforms):
+            if isinstance(transform, Parallel):
+                self.connect_graph(self.input_node,
+                                   transform,
+                                   output_slice=idx)
+                out_nodes = transform.parents[transform.output_node]
+                out_node = out_nodes[idx]
+                self.connect(out_node,
+                             self.output_node)
+                continue
+            if isinstance(transform, Pipeline):
+                self.connect_graph(self.input_node,
+                                   transform,
+                                   output_slice=idx,
+                                   parent_position=None)
+                out_nodes = transform.parents[transform.output_node]
+                for out_node in out_nodes:
+                    self.connect(out_node,
+                                 self.output_node)
+                continue
```

Comment:
Is it possible to write these tasks on different methods in `Parallel` and `Pipeline` so we don't make `isinstance` checks in here and repeat code?